src/test/java/org/jenkinsci/test/acceptance/recorder/TestRecorderRuleTest.java

```diff
@@ -0,0 +1,106 @@
+package org.jenkinsci.test.acceptance.recorder;
+
+import org.hamcrest.core.Is;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TestWatcher;
+import org.junit.runner.Description;
+
+import java.io.File;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+
+public class TestRecorderRuleTest {
+
+    @Test
+    public void shouldNotRecordSuccessTestExecutionByDefault() {
+        TestRecorderRule testRecorderRule = new TestRecorderRule();
+        Description shouldNotRecordSuccessTestExecutionByDefault = Description.createTestDescription("org.jenkinsci.test.acceptance.recorder.TestRecorderRuleTest", "shouldNotRecordSuccessTestExecutionByDefault");
+        testRecorderRule.starting(shouldNotRecordSuccessTestExecutionByDefault);
+
+        System.out.println("Hello World");
+
+        testRecorderRule.succeeded(shouldNotRecordSuccessTestExecutionByDefault);
+        testRecorderRule.finished(shouldNotRecordSuccessTestExecutionByDefault);
+
+        File outputFile = new File("target", "org.jenkinsci.test.acceptance.recorder.TestRecorderRuleTest-shouldNotRecordSuccessTestExecutionByDefault.mov");
+        assertThat(outputFile.exists(), is(false));
+
+        //Clean the field
+        outputFile.delete();
+    }
+    @Test
+    public void shouldRecordFailingTestExecutionByDefault() {
+
+        TestRecorderRule testRecorderRule = new TestRecorderRule();
+        Description shouldNotRecordSuccessTestExecutionByDefault = Description.createTestDescription("org.jenkinsci.test.acceptance.recorder.TestRecorderRuleTest", "shouldRecordFailingTestExecutionByDefault");
+        testRecorderRule.starting(shouldNotRecordSuccessTestExecutionByDefault);
+
+        System.out.println("Good Bye World");
+        //succeeded is not called since a failure is simulated
+        //testRecorderRule.succeeded(shouldNotRecordSuccessTestExecutionByDefault);
+        testRecorderRule.finished(shouldNotRecordSuccessTestExecutionByDefault);
+
+        File outputFile = new File("target", "org.jenkinsci.test.acceptance.recorder.TestRecorderRuleTest-shouldRecordFailingTestExecutionByDefault.mov");
+        assertThat(outputFile.exists(), is(true));
+
+        //Clean the field
+        outputFile.delete();
+    }
+
+    @Test
+    public void shouldRecordSuccessTestExecutionWhenSaveAll() {
+        String oldValue = System.getProperty("RECORDER_SAVE_ALL");
+        System.setProperty("RECORDER_SAVE_ALL", "true");
```

Comment:
You prefer to have the recording feature disabled by default. But I think that since it is for ATH it will always interesting having this information. If we disabled by default, then it will be difficult to remember to enable it. Also it can happens that just when these random fails happens you had the recording disabled because you didn't remember to set the property.